services/cloudapi-v6/resources/image.go

```diff
@@ -41,6 +64,47 @@ func NewImageService(client *Client, ctx context.Context) ImagesService {
 	}
 }
 
+func (s *imagesService) Upload(p UploadProperties) error {
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: true, // TODO: INSECURE. Change this before prod! Client susceptible to "Man-in-the-middle" attacks.
+	}
+
+	dialOptions := ftps.DialOptions{
+		Host:        p.Url,
+		Port:        p.Port,
+		Username:    viper.GetString(config.Username),
+		Passowrd:    viper.GetString(config.Password),
+		ExplicitTLS: true,
+		TLSConfig:   &tlsConfig,
+	}
+
+	//ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	//defer cancel()
+
+	c, err := ftps.Dial(context.TODO(), dialOptions)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Connected to %s\n", p.Url)
+
+	// Do something with the FTP conn
+	err = c.Chdir(filepath.Dir(p.Path))
+	if err != nil {
+		fmt.Printf("Failed to change to %s\n", filepath.Dir(p.Path))
+		return err
+	}
+
+	data := bytes.NewBufferString("Hello World")
+	err = c.Upload(s.context, "test.iso", data)
```

Comment:
s.context is nil that is the reason for the panic.

It would be good if we can default the name of the uploaded file to the basename of the file to upload, but it would be very handy if this uploaded name can be overwritten. This is especially true for CI pipelines, where always the same image name comes out of the pipeline but the upload name should be different all the time.

Reply:
Ah, good find, thanks.

I was so focused on the context passed to "Dial" function that this other context slipped unnoticed

As for the auto basename, I had already been doing that (see prev. commits) but decided to temporarily always have the same image name for debugging purposes. However, the suggestion for having a customizable name via a flag is a great idea! 

Thanks!